packages/amplify-provider-awscloudformation/src/aws-utils/aws-cognito-client.ts

```diff
@@ -0,0 +1,27 @@
+import { $TSContext } from 'amplify-cli-core';
+import aws from './aws.js';
+import * as AWS from 'aws-sdk';
+import { AwsSecrets, loadConfiguration } from '../configuration-manager';
+
+export class CognitoUserPoolClient {
```

Comment:
this should be called `CognitoUserPoolClientFactory` or `CognitoUserPoolClientProvider` by looking at implementation.